src/IIIFPresentation/API/Features/Storage/Requests/PostHierarchicalCollection.cs

```diff
@@ -52,7 +53,9 @@ public async Task<ModifyEntityResult<Collection, ModifyCollectionType>> Handle(P
         var parentCollection =
             await dbContext.RetrieveHierarchy(request.CustomerId, parentSlug, cancellationToken);
         
-        if (parentCollection == null) return ErrorHelper.NullParentResponse<Collection>();
+        var parentValidationError =
+            ParentValidator.ValidateParentCollection<Collection>(parentCollection.Collection);
```

Comment:
Is `parentCollection` guaranteed to have a value? Should this be `parentCollection?.Collection` or validate that `parentCollection != null` before passing?

This also differs from `CreateCollection` and `ManifestWriteService` in that it calls `dbContext.RetrieveHierarchy` to get parent, rather than `dbContext.RetrieveCollectionAsync` - why the difference?